File: scratch/extract.py
import io

with io.open('terminal/terminal-tasks.js', 'r', encoding='utf-8') as f:
    content = f.read()

start = content.find('async function loadTasks(isSilent = false) {')
end = content.find('function renderTasks(tasks) {')

logic = content[start:end]

with io.open('scratch/extracted_raw.js', 'w', encoding='utf-8') as f:
    f.write(logic)

File: scratch/extract.js
const fs = require('fs');

let terminalJS = fs.readFileSync('terminal/terminal-tasks.js', 'utf8');

let startLoadTasks = terminalJS.indexOf('async function loadTasks(isSilent = false) {');
let endLoadTasks = terminalJS.indexOf('function renderTasks(tasks) {');
let logicToExtract = terminalJS.substring(startLoadTasks, endLoadTasks);

fs.writeFileSync('scratch/extracted_raw.js', logicToExtract);
